CMakeLists.txt

```diff
@@ -2,16 +2,22 @@
 cmake_minimum_required(VERSION 2.8)
 project(opensplice)
 
-# Hacky, but will work for now
-math(EXPR BITS "8*${CMAKE_SIZEOF_VOID_P}")
+include (TargetArch.cmake)
+target_architecture(ARCH)
+message(STATUS "Building for arch: ${ARCH}")
+
 if(APPLE)
   set(platform "x86_64.darwin10_clang-release")
 elseif(WIN32)
   message(FATAL_ERROR "Windows not yet supported")
 else()
-  if (BITS EQUAL 64)
+  if (ARCH EQUAL "x86_64")
     set(platform "x86_64.linux-release")
-  else (BITS EQUAL 64)
+  elseif (ARCH EQUAL "armv7")
+    set(platform "armv7l.linux-release")
+  elseif (ARCH EQUAL "aarch64") # aarch64 works with armv7l profile
+    set(platform "armv7l.linux-release")
+  else (ARCH EQUAL "i386")
```

Comment:
This is an `else` statement, I assume the `ARCH EQUAL "i386"` argument is just documentation?